src/app/ApplicationTemplate.Shared.Views/Framework/Browser/DispatcherLauncherServiceDecorator.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using System.Reactive;
+using System.Threading.Tasks;
+using Microsoft.UI.Dispatching;
+
+namespace ApplicationTemplate;
+
+/// <summary>
+/// Invokes calls to an <see cref="ILauncherService"/> on the <see cref="DispatcherQueue"/>.
+/// </summary>
+public class DispatcherLauncherServiceDecorator : ILauncherService
```

Comment:
This works, but now that we own the implementation, shouldn't we just deal with the dispatcher directly in the implementation?

Reply:
Yes it does make sense I could do that.